connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -536,20 +561,37 @@ private void processConnectorConfigUpdates(Set<String> connectorConfigUpdates) {
         // If we only have connector config updates, we can just bounce the updated connectors that are
         // currently assigned to this worker.
         Set<String> localConnectors = assignment == null ? Collections.<String>emptySet() : new HashSet<>(assignment.connectors());
+        log.trace(
+            "Processing connector config updates; "
+                + "currently-owned connectors are {}, and to-be-updated connectors are {}",
+            localConnectors,
+            connectorConfigUpdates
+        );
         for (String connectorName : connectorConfigUpdates) {
-            if (!localConnectors.contains(connectorName))
+            if (!localConnectors.contains(connectorName)) {
+                log.trace(
+                    "Skipping config update for connector {} as it is not owned by this worker",
+                    connectorName
+                );
                 continue;
+            }
             boolean remains = configState.contains(connectorName);
             log.info("Handling connector-only config update by {} connector {}",
                     remains ? "restarting" : "stopping", connectorName);
-            worker.stopConnector(connectorName);
+            worker.stopAndAwaitConnector(connectorName);
             // The update may be a deletion, so verify we actually need to restart the connector
             if (remains)
-                startConnector(connectorName);
+                startConnector(connectorName, (error, result) -> { });
```

Comment:
The caller of this function doesn't need to block on starting the connector, but I think it should log errors inside of the callback. Otherwise they're going to get swallowed.

Reply:
Agh, you're right. Thought I'd committed that but apparently not. Will add.